extensions.md

```diff
@@ -186,3 +186,53 @@ the server, but LSP does not expose them.
   - Clients that don't support client-side filtering should ignore this and
     use `sortText`, which incorporates `score` and fuzzy-matching.
 
+## AST
+{:v12}
+
+C++ has a complicated grammar and semantic structure that's not always obvious
+from the source. Inspecting the clang AST can lend some insight.
+Despite the name, it captures semantics as well as syntax (e.g. implicit casts).
+
+**New client->server request**: `textDocument/ast`:
+ - Params: `ASTParams` object with properties:
+   - `textDocument : TextDocumentIdentifier`: the open file to inspect
+   - `range : Range`: identifies the AST node to fetch.
+     The highest-level node that entirely contains the range is returned.
+ - Result: `ASTNode` object with properties:
+   - `role : string`: the general kind of node, such as "expression".
```

Comment:
Is there any list of roles? I was thinking using Clang class names was logical otherwise the mapping is not obvious.

Reply:
I've added the most common (expression/statement/type/declaration).
I deliberately don't want this to be a reference for all possible values - it's a maintenance trap, it's subject to change, there's lots of obscure cases... and this is *not* an enum by design, the idea/hope is that other language servers could implement it too.

The VSCode extension shows how clients can use these values - display them, add optional behavior (like icons) for a few known values but with graceful fallback.